CONTRIBUTING.rst

```diff
@@ -28,88 +50,87 @@ Rasterio's API is different from GDAL's API and this is intentional.
+to give other developers a heads up and to provide an opportunity for valuable
+early feedback.
+
 Code Conventions
 ================
 
 The ``rasterio`` namespace contains both Python and C extension modules. All
 C extension modules are written using `Cython <http://cython.org/>`__. The
 Cython language is a superset of Python. Cython files end with ``.pyx`` and
-``.pxd`` and are where we keep all the code that calls C functions in the GDAL
-library.
+``.pxd`` and are where we keep all the code that calls GDAL's C functions.
 
-Rasterio supports Python 2 and Python 3 in the same code base. We have a
-compatibility module named ``five.py``. Renaming it is a to-do.
+Rasterio supports Python 2 and Python 3 in the same code base, which is
+aided by an internal compatibility module named ``five.py``. Renaming it is a
+to-do (`#719 <https://github.com/mapbox/rasterio/issues/719>`__).  It functions
+similarly to the more widely known `six <https://pythonhosted.org/six/>`__ but
+we only use a small portion of the features so it eliminates a dependency.
 
 We strongly prefer code adhering to `PEP8
 <https://www.python.org/dev/peps/pep-0008/>`__.
 
-Tests are mandatory for new features. We use ``pytest``. Tests written using
-``unittest`` are fine, too.
-
-We aspire to 100% coverage for Python modules. Coverage of the Cython code is
-a future aspiration.
+Tests are mandatory for new features. We use `pytest <https://pytest.org>`_,
+although those written using the builtin `unittest
+<https://docs.python.org/3/library/unittest.html>`_ module are fine, too.
 
-Git Conventions
-===============
+We aspire to 100% coverage for Python modules but overage of the Cython code is
+a future aspiration (`#515 <https://github.com/mapbox/rasterio/issues/515>`__).
 
-TODO
-
-Issue Conventions
-=================
-
-Rasterio is a relatively new project and highly active. We have bugs, both
-known and unknown.
-
-Please do a search of existing issues, open and closed, before creating a
-new one.
-
-Because Rasterio has C extension modules, bug reports very often hinge on the
-following details:
-
-- Operating system type and version (Windows? Ubuntu 12.04? 14.04?)
-- The source of the Rasterio distribution you installed (PyPI, Anaconda, or
-  somewhere else?)
-- The version and source of the GDAL library on your system (UbuntuGIS? 
-  Homebrew?)
+Development Environment
+=======================
 
-Please provide these details as well as Rasterio version and tracebacks from
-your program. Short scripts and datasets that demonstrate the bug are 
-especially helpful!
+Developing Rasterio requires Python 2.7 or any final release after and
+including 3.4.  We prefer developing with the most recent version of Python
+but recognize this is not possible for all contributors.  A C compiler is also
+required to leverage `existing protocols
+<https://docs.python.org/3.5/extending/extending.html>`__ for extending Python
+with C or C++.  Windows compiler information is forthcoming.
```

Comment:
@brendan-ward I'll defer to your Windows knowledge.  If our users have to jump through those hoops to install Rasterio then I think we should favor them in the documentation rather than the developers and point the devs to the readme.